ComSemApp/student/views.py

```diff
@@ -799,17 +793,23 @@ def get_context_data(self, **kwargs) -> dict[str, dict[str, str | int | float]]:
         """
         context : dict[str, Any] = super(SpeakingPracticeResultsView, self).get_context_data(**kwargs)
         context['attempts'] : list[dict[str, str | float]] = {}
-        try:
-            attempt_ids : list[str] = dict(self.request.GET)['attempt_ids']
-        except:
-            return context
-        attempts : QuerySet[SpeakingPracticeAttempt] = SpeakingPracticeAttempt.objects.filter(student=self.student, pk__in=attempt_ids)
+        attempt_ids : list[str] = dict(self.request.GET).get('attempt', None)
+        session_ids : list[str] = dict(self.request.GET).get('session', None)
+
+        # if both are null
+        if(not (attempt_ids or session_ids)):
+            raise BadRequest("No attempt id or session id given")
+        
+        attempts : QuerySet[SpeakingPracticeAttempt] = SpeakingPracticeAttempt.objects.filter(
+            Q(session__student=self.student)
+            & Q(worksheet__course=self.course)
```

Comment:
I got an error here. attempts don't have worksheets. need expression=>worksheet=> course.

after editing this I am getting another error:
'NoneType' object is not iterable

It might regard what I wrote below

Reply:
Fixed. It's a bit funky but I think it's the best way to do it.